R/mock-session.R

```diff
@@ -1,3 +1,31 @@
+# Promise helpers taken from:
+#   https://github.com/rstudio/promises/blob/master/tests/testthat/common.R
+# Block until all pending later tasks have executed
+wait_for_it <- function() {
+  while (!later::loop_empty()) {
+    later::run_now()
```

Comment:
I'm pretty sure the `run_now` and `Sys.sleep` can be replaced with `run_now(0.1)`. I'm not sure why it was written that way in promises. @jcheng5 can you enlighten us?

Reply:
I’m sure you’re right. I just threw those helpers together.